src/plugins/intel_npu/src/plugin/npuw/base_sync_infer_request.cpp

```diff
@@ -495,9 +492,45 @@ void ov::npuw::IBaseInferRequest::handle_quant_host_gather(std::size_t idx, RqPt
             return ov::Shape{1, ids_shape[1], shape.size() == 3 ? shape[1] * shape[2] : shape[1]};
         };
 
-        ov::Tensor gatherw(vocabw->get_element_type(), get_gathered_shape(vocabw->get_shape()));
+        m_quant_gather_tensors.w = ov::Tensor(vocabw->get_element_type(), get_gathered_shape(vocabw->get_shape()));
+
+        if (comp_model_desc.quant_unpack_gather.src_z_idx != -1 &&
+            comp_model_desc.quant_unpack_gather.src_s_idx != -1) {
```

Comment:
I think if you take a reference to `quant_unpack_gather` once in the beginning of this method, like with `comp_model_desc`, it will make the rest of the code less verbose.